src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -6117,16 +6118,21 @@ private void LoadSearchValues(ReadOnlySpan<char> chars)
             // Logically do _searchValues[index], but avoid the bounds check on accessing the array,
             // and cast to the known derived sealed type to enable devirtualization.
 
-            // DerivedSearchValues d = Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(this._searchValues), index);
+            // DerivedSearchValues d = Unsafe.As<DerivedSearchValues>(Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(this._searchValues), index));
             // ... = d;
             Ldthisfld(s_searchValuesArrayField);
             Call(s_memoryMarshalGetArrayDataReferenceSearchValues);
             Ldc(index * IntPtr.Size);
             Add();
             _ilg!.Emit(OpCodes.Ldind_Ref);
-            LocalBuilder ioavLocal = _ilg!.DeclareLocal(list[index].GetType());
-            Stloc(ioavLocal);
-            Ldloc(ioavLocal);
+            Call(MakeUnsafeAs(list[index].GetType())); // provide JIT with details necessary to devirtualize calls on this instance
+
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2060:MakeGenericMethod", Justification =
+                "Calling Unsafe.As<T> is safe since the T doesn't have trimming annotations.")]
+            static MethodInfo MakeUnsafeAs(Type type)
+            {
+                return Unsafe.As<MethodInfo>(s_unsafeAs.MakeGenericMethod(type));
```

Comment:
```suggestion
                return s_unsafeAs.MakeGenericMethod(type);
```

![image](https://github.com/dotnet/runtime/assets/13110571/a49225f6-f336-4ac8-b168-76e2c815b275)

Reply:
This is from when I thought the complaint was that the method might not be rooted. The previous justification explained its presence.

Reply:
(The image macro was a reference to https://knowyourmeme.com/memes/xzibit-yo-dawg; looking at this again it might not be obvious to people who didn't spend unreasonable amounts of time on imgur in the 2010s.)